### PR TITLE
[IMP] hr_timesheet: use name_get to add employee's company

### PR DESCRIPTION
**Pull request by ppr-odoo**
1. hr_timesheet: use name_get to add employee's company

   Currently, if both companies have employees with the same name, it is not known which
company the selected employee belongs to.

    In  this commit, the name of the company will only be appended by the name_get when more than
      one company is present in the database.

   Example
    - Administrator - My Company
    - Administrator - Tiny ERP

task-3101249

### CODE REVIEW

File: addons/hr_timesheet/models/hr_employee.py [hunk 1]
```diff
@@ -7,6 +7,22 @@
...
+        res = super().name_get()
+        if len(self.env.context.get('allowed_company_ids', [])) <= 1:
+            return res
+        name_mapping = dict(res)
+        employee_read_group = self.env['hr.employee'].sudo()._read_group(
+            [('user_id', 'in', self.user_id.ids)],
+            ['user_id'],
+            ['user_id'],
+        )
+        employees_count_per_user = {res['user_id'][0]: res['user_id_count'] for res in employee_read_group}
```
Comment (auon-odoo) [inline addons/hr_timesheet/models/hr_employee.py:20]:
```suggestion
        employees_count_per_user = {
            res['user_id'][0]: res['user_id_count']
            for res in self.env['hr.employee'].sudo()._read_group(
                [('user_id', 'in', self.user_id.ids)],
                ['user_id'],
                ['user_id'],
            )
        }
```
Small detail number 1 :angel:

File: addons/hr_timesheet/views/hr_timesheet_portal_templates.xml [hunk 2]
```diff
@@ -118,8 +118,15 @@
...
-                                    <td t-if="not groupby == 'task'"><span t-field="timesheet.task_id" t-att-title="timesheet.task_id.display_name"/></td>
+                                    <td t-if="not groupby == 'project'">
+                                        <a t-attf-href="#{timesheet.project_id.access_url}" t-out="timesheet.project_id.display_name"
+                                            t-att-title="timesheet.project_id.display_name" t-if="timesheet.project_id.id in access_project_ids"/>
+                                        <t t-else="" t-out="timesheet.project_id.display_name" t-att-title="timesheet.project_id.display_name"/>
+                                    </td>
+                                    <td t-if="not groupby == 'task'">
+                                        <a t-attf-href="#{timesheet.task_id.access_url}" t-out="timesheet.task_id.display_name"
+                                            t-att-title="timesheet.task_id.display_name"/>
+                                    </td>
```
Comment (auon-odoo) [inline addons/hr_timesheet/views/hr_timesheet_portal_templates.xml:129]:
```suggestion
                                    <td t-if="not groupby == 'project'">
                                        <a  t-if="timesheet.project_id.id in access_project_ids" t-out="timesheet.project_id.display_name"
                                            t-att-title="timesheet.project_id.display_name" t-attf-href="#{timesheet.project_id.access_url}"/>
                                        <t t-else="" t-out="timesheet.project_id.display_name" t-att-title="timesheet.project_id.display_name"/>
                                    </td>
                                    <td t-if="not groupby == 'task'">
                                        <a t-out="timesheet.task_id.display_name" t-att-title="timesheet.task_id.display_name"
                                            t-attf-href="#{timesheet.task_id.access_url}"/>
                                    </td>
```
Small detail number 2. `t-if` first, and the rest is just keeping consistency in attrs's order. :angel:

File: addons/sale_timesheet/views/sale_timesheet_portal_templates.xml [hunk 3]
```diff
@@ -87,7 +87,11 @@
             <th t-if="not groupby == 'sol'">Sales Order Item</th>
         </xpath>
         <xpath expr="//tbody//td[hasclass('text-end')]" position="before">
-            <td t-if="not groupby == 'sol'"><span t-field="timesheet.so_line" t-att-title="timesheet.so_line.display_name"></span></td>
+            <td t-if="not groupby == 'sol'">
+                <a t-attf-href="#{timesheet.order_id.access_url}" t-out="timesheet.so_line.display_name"
+                    t-att-title="timesheet.so_line.display_name" t-if="timesheet.order_id.id in access_sale_order_ids"/>
+                <t t-else="" t-out="timesheet.so_line.display_name" t-att-title="timesheet.so_line.display_name"/>
```
Comment (auon-odoo) [inline addons/sale_timesheet/views/sale_timesheet_portal_templates.xml:93]: Same here :angel:

**Review by auon-odoo — CHANGES_REQUESTED** [review]
Thank you for this super neat work! :fire: 
Few details and we'll be good to go!

**Review by auon-odoo — APPROVED** [review]
Thank you :fire:

**Review by xavierbol — APPROVED** [review]
LGTM, thanks for your work! 🙂 

robodoo r+

**Pull request closed without merging**